usecases/decision_usecase.go

```
@@ -450,11 +452,16 @@ func (usecase *DecisionUsecase) CreateDecision(
 		}
 
 		if decision.SanctionCheckExecution != nil {
-			if _, err := usecase.sanctionCheckRepository.InsertSanctionCheck(ctx, tx,
-				decision.DecisionId, *decision.SanctionCheckExecution, true); err != nil {
+			sc, err := usecase.sanctionCheckRepository.InsertSanctionCheck(ctx, tx,
+				decision.DecisionId, *decision.SanctionCheckExecution, true)
+			if err != nil {
 				return models.DecisionWithRuleExecutions{},
 					errors.Wrap(err, "could not store sanction check execution")
 			}
+
+			if usecase.openSanctionsRepository.IsSelfHosted(ctx) {
+				_ = usecase.taskQueueRepository.EnqueueMatchEnrichmentTask(ctx, input.OrganizationId, sc.Id)
```

let's still return the error, no ? Even if it's a secondary action

---

(I can perhaps be converted to just logging it)

---

Yes, I actually wanted to log and ignore the error, but forgot. I don't think the whole decision should fail if it fails to be enriched automatically, since it can always be enriched after the fact. I'll add the logging if you agree.

---

yep, let's do that 👍  (you can use the `utils.LogAndReportSentryError(ctx, err)` function which, well, logs and reports to sentry)